src/main/java/club/minnced/discord/webhook/send/component/PartialEmoji.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018-2020 Florian Spie�
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package club.minnced.discord.webhook.send.component;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONObject;
+import org.json.JSONString;
+
+public class PartialEmoji implements JSONString {
+
+	private final String name;
+	private final long id;
+	private final boolean animated;
+
+	private PartialEmoji(@NotNull String name, @NotNull String id, boolean animated) {
+		this(name, Long.parseLong(id), animated);
+	}
+
+
+	private PartialEmoji(@NotNull String name, long id, boolean animated) {
+		this.name = name;
+		this.id = id;
+		this.animated = animated;
+	}
+
+	/**
+	 * Emoji with only necessary features required by webhooks
+	 * @param name
+	 *        the name of the emoji
+	 * @param id
+	 *        the id of the emoji
+	 * @param animated
+	 *        true if the emoji is animated
+	 * @return An instance of Partial emoji with the provided info
+	 */
+	@NotNull
+	public static PartialEmoji of(@NotNull String name, long id, boolean animated) {
+		return new PartialEmoji(name, id, animated);
+	}
+
+	/**
+	 * Emoji with only necessary features required by webhooks
+	 * @param name
+	 *        the name of the emoji
+	 * @param id
+	 *        the id of the emoji
+	 * @param animated
+	 *        true if the emoji is animated
+	 * @return An instance of Partial emoji with the provided info
+	 */
+	@NotNull
+	public static PartialEmoji of(@NotNull String name, String id, boolean animated) {
```

Comment:
```suggestion
	public static PartialEmoji of(@NotNull String name, @NotNull String id, boolean animated) {
```

Reply:
You should make an explicit distinction between unicode and custom emoji through named factory methods:

```java
Emoji.custom(id, name, animated)
Emoji.unicode(code)
```